src/Mesh/getEdgeIntegralOfPolygonalChain.jl

```diff
@@ -0,0 +1,234 @@
+export getEdgeIntegralOfPolygonalChain
+
+using Base.BLAS
+
+"""
+        function jInv.Mesh.getEdgeIntegralOfPolygonalChain
+
+        s = getEdgeIntegralPolygonalChain(mesh,polygon)
+        s = getEdgeIntegralPolygonalChain(mesh,polygon,normalize)
+        
+        Compute the integral of a piecewise linear edge grid basis projected onto
+        the edges of a polygonal chain. This function can be used to evaluate the
+        source term of a line current carried by the polygonal chain in electromagnetic
+        modelling.
+        
+        The piecewise linear edge grid basis consists of the functions
+          phix_i (i = 1 ... numXEdges)
+          phiy_i (i = 1 ... numYEdges)
+          phiz_i (i = 1 ... numZEdges)
+        where phix_i, phiy_i, phiz_i = 1 at the i-th x/y/z-directed edge
+          and phix_i, phiy_i, phiz_i = 0 at all other edges.
+        
+        INPUT
+        mesh ...... OcTree mesh
+        polygon ... vertices of polygonal chain as numVertices x 3 array
+        normalize . divide line integral by length of integration path (boolean, optional)
+        
+        OUTPUT
+        s ......... source vector
+        
+        For a closed current loop, specify polygon such that
+        polygon[1,:] == polygon[end,:]
+"""
+
+function getEdgeIntegralOfPolygonalChain(M::TensorMesh3D,polygon::Array{Float64,2};
```

Comment:
Do you think this method can be extended for other meshes (i.e. include RegularMesh or OcTree Mesh)?